go/vt/vttablet/tabletserver/planbuilder/plan_test.go

```diff
@@ -106,10 +110,11 @@ func TestPlanPoolUnsafe(t *testing.T) {
 			var err error
 			statement, err := sqlparser.Parse(tcase.input)
 			require.NoError(t, err)
-			// In Pooled Connection, plan building will fail.
-			plan, err = Build(statement, testSchema, false /* isReservedConn */, "dbName")
-			require.Error(t, err)
-			out := err.Error()
+			// Plan building will not fail, but it will mark that reserved connection is needed.
+			plan, err = Build(statement, testSchema, "dbName")
+			require.NoError(t, err)
+			require.True(t, plan.NeedsReservedConn)
+			out := fmt.Sprintf("%s not allowed without reserved connection", plan.PlanID.String())
 			if out != tcase.output {
```

Comment:
If there is no error here from the Build step, why are we asserting on a error output stored in the file? The error doesn't come from the codebase, we are checking on something that is present in the test files only

Reply:
The plan is allowed, but the execution will fail. I felt it makes sense to put the error message in file and recreate here but I think may be it is not important and move this test to place where it actually returns this error.